File: 1544-make-the-string-great/1544-make-the-string-great.cpp
class Solution {
public:
    string makeGood(string s) {
        // Use stack to store the visited characters.
        vector<char> stack;
        
        // Iterate over 's'.
        for (auto currChar : s) {
            // If the current character make a pair with the last character in the stack,
            // remove both of them. Otherwise, we add the current character to stack.
            if (!stack.empty() && abs(stack.back() - currChar) == 32)
                stack.pop_back(); 
            else
                stack.push_back(currChar);
        }
        
        // Returns the string concatenated by all characters left in the stack.
        string ans(stack.begin(), stack.end());
        return ans;
    }
};

File: 1544-make-the-string-great/1544-make-the-string-great.py
class Solution:
    def makeGood(self, s: str) -> str:
        # if s has less than 2 characters, we just return itself.
        while len(s) > 1:
            # 'find' records if we find any pair to remove.
            find = False
            
            # Check every two adjacent characters, curr_char and next_char.
            for i in range(len(s) - 1):
                curr_char, next_char = s[i], s[i + 1]
                
                # If they make a pair, remove them from 's' and let 'find = True'.
                if abs(ord(curr_char) - ord(next_char)) == 32:
                    s = s[:i] + s[i + 2:]
                    find = True
                    break
            
            # If we cannot find any pair to remove, break the loop. 
            if not find:
                break
        return s